sotodlib/calibration/calibration_tod_analysis.py

```diff
@@ -0,0 +1,749 @@
+
+    if rank==0:
+        full_df = pd.concat(all_dfs)
+        full_df.dropna()
+        full_df = full_df.set_index(full_df['det_name'])
+
+        ## Convert to structured array and store file - minor problem with string encoding
+        #dt = h5py.special_dtype(vlen=str) 
+        new_dtypes = {"amp": np.float64, "xi0": np.float64, "eta0":np.float64, "fwhm_xi":np.float64, \
+                      "fwhm_eta":np.float64, "phi":np.float64, "tau":np.float64, "det_name":str, "snr":np.float64}
+        full_df = full_df.astype(new_dtypes)
+        
+        # calculating relative and absolute calibration
+        amp = full_df.amp.values
+        full_df['rel_cal'] = amp/np.mean(amp)
+
+        beam_file = '/mnt/so1/shared/site-pipeline/bcp/%s_%s_beam.h5'%(tele, band)
+        sso_obj = sim_sso.OpSimSSO(sso_name, beam_file)
+
+        sso_obj._get_planet_temp(sso_name)
+        datestr = '%s-%02d-%02d'%(year, int(month), int(day))
+        sso_obj.sso.compute(datestr)
+        ttemp = np.interp(float(band[1:]), sso_obj.t_freqs, sso_obj.ttemp)
+        beam, _ = sso_obj._get_beam_map(sso_obj.sso.size, ttemp)
+        amp_ref = beam(0, 0)[0][0]
+        full_df['abs_cal'] = amp/amp_ref
+        
+        ### Again need to decide on the naming convention
+        obs_id = '%s_%s_%s_%s_%s_%s-%02d-%02d'%(tele, tube, band, wafer, sso_name, year, int(month), int(day))
+        h_name = 'fitting_table_'+obs_id
+
+        #hf = h5py.File(opj(outdir, h_name), 'w')
+        #hf.create_dataset('fitted_params_'+obs_id, data=summary_data) 
+        #hf.close() 
+
+        ## Assert tables dependnecy is properly satisfied
+        out_folder = opj(outdir, tele, tube, wafer)        
+        if not os.path.exists(out_folder):
+            # Create a new directory because it does not exist 
+            os.makedirs(out_folder)
+        full_df.to_hdf(opj(out_folder, h_name+'.h5'),key='parameter_table', mode='w')
+
+        t2 = time.time()
+        print('Time to run fittings for a full wafer is {} seconds'.format(t2-t1))
+
+        ## check why the mapmaker was working only with context and fix this issue
+        if map_make and f_format=='context' and n_modes is not None:
+            beamsize = np.degrees(init_params[2])/2 
+            make_maps(path, outdir, obs_id, sso_name, beamsize, n_modes)
+
+
+
+def main():
+    
+    parser = ap.ArgumentParser(formatter_class=\
+    ap.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--indir', action='store', dest='indir',
+    help='The location of the simulated files.')
+    parser.add_argument('--outdir', action='store', dest='outdir',
+    help='The location for the .h5 ouput files to be stored.')
+    parser.add_argument('--f_format', action='store', dest='f_format',
+    help='The format of the simulations [g3, h5, context].')
+    parser.add_argument('--tele', action='store', dest='tele',
+    help='The telescope name [LAT, SAT].')
+    parser.add_argument('--tube', action='store', dest='tube',
```

Comment:
I will say that as someone who does LATR instrument work, I find knowing what tube something is associated with to be very useful since that is the basis that we are most familiar with.

Reply:
Then do you suggest we go from LAT(SAT) directly to wafer number? Do we have some examples of ObsDb? I think if tube number can be easily found there, we can perhaps be more comfortable dropping tube here.